services/app-api/postgres/submissionWithRevisionsHelpers.ts

```diff
@@ -45,11 +45,11 @@ function convertToSubmission2Type(dbSub: StateSubmissionWithRevisions): Submissi
         stateCode: dbSub.stateCode,
         revisions: dbSub.revisions.map(r => { 
             let submitInfo: UpdateInfoType | undefined = undefined
-            if (r.submittedAt) {
+            if (r.submittedAt && r.submittedReason && r.submittedBy) {
```

Comment:
Just to note, this is somewhere where our db representation and our domain model representation differ, and it means that we can get the two into an inconsistent state. This code means that if we ever manage to write incomplete submit* fields to the database, we’ll end up with no submitInfo coming out. I think that’s correct, but I want to point it out because it’s a bit of a risk. If we were really worried about that happening, we could raise an error here instead